app/components/file-uploader.js

```diff
@@ -145,45 +196,72 @@ export default Ember.Component.extend({
             }, 1500);
             return this.get('callback.promise');
         },
+        discardUploadChanges() {
+            // Discards file upload changes.  Removes staged files from Dropzone, reverts visible file version.
+            if (window.Dropzone) window.Dropzone.forElement('.dropzone').removeAllFiles(true);
+            this.set('fileVersion', this.get('osfFile.currentVersion') || 1);
+            this.attrs.discardUploadChanges();
+        },
         maxfilesexceeded(_, dropzone, file) {
             dropzone.removeAllFiles();
             dropzone.addFile(file);
         },
         complete(_, dropzone, file) {
+            // Called after uploading file to node is complete. After file has been uploaded,
+            // makes a request to create the preprint.
+
             // Complete is called when swapping out files for some reason...
             if (file.xhr === undefined) return;
 
             if (Math.floor(file.xhr.status / 100) === 2) {
-                // Success
-                this.get('toast').info('Preprint uploaded!');
-
+                // File upload success
                 let resp = JSON.parse(file.xhr.response);
                 this.get('store')
                     .findRecord('file', resp.data.id.split('/')[1])
-                    .then(file => {
-                        this.set('osfFile', file);
-                        this.sendAction('finishUpload');
-                    });
+                        .then(file => {
+                            this.set('osfFile', file);
+                            if (this.get('nodeLocked')) { // Edit mode
+                                this.get('toast').info('Preprint file updated!');
```

Comment:
If you can set the text via handlebars template, that would be ideal e.g.

```hbs
{{if nodeLocked (t 'path.to.translation.obj')}}
```

Otherwise, as long as it can be changed based on the language (i.e. not just importing the english translation and using that) then it's all the same.

Looking at the docs, [Ember-i18n provides a macro to do it](https://github.com/jamesarosen/ember-i18n/wiki/Doc:-Translating-Text#translation-computed-property-macro).